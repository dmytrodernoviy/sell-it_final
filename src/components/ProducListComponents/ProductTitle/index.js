import React from 'react'
import ProductTitleStyle from './style';
import FaEye from '../../../assets/img/fa-eye.png'
import {Link} from 'react-router-dom'

const ProductTitle = ({id, title}) => 
    <ProductTitleStyle>
        <span>{title}</span>
        <Link to={`/product/${id}`}>
            <img src={FaEye} alt="FaEye" />
        </Link>
    </ProductTitleStyle>

export default ProductTitle