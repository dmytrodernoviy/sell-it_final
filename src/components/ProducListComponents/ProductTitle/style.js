import styled from 'styled-components'

const ProductTitleStyle = styled.div`
    width: 100%;
    background: #00b3be;
    display: flex;
    align-items: center;
    justify-content: space-between;
    color: white;
    font-size: 18px;
    padding: 10px;
`

export default ProductTitleStyle