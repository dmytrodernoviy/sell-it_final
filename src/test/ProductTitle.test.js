import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProductTitle from '../components/ProducListComponents/ProductTitle';

Enzyme.configure({ adapter: new Adapter() });

const props = {
    id: 244,
    title: ""
}

describe ('ProductTitle', () => {
    it("ProductTitle initial", () => {
        const Component = shallow(<ProductTitle {...props}/>);
        expect(Component.find("span")).toHaveLength(1);
        expect(Component.find("img")).toHaveLength(1);
        expect(Component.find("Link")).toHaveLength(1);
    })
    
    it("ProductTitle change title property", () => {
        const changedProps = {
            ...props,
            title: "Продам апельсинку"
        }

        const Component = shallow(<ProductTitle {...changedProps} />)
        expect(Component.find("span").text()).toEqual("Продам апельсинку")
    })

    it("ProductTitle Link attribute 'to' ", () => {
        const Component = shallow(<ProductTitle {...props} />) 
        expect(Component.find("Link").prop("to")).toEqual('/product/244')
        expect(Component).toMatchSnapshot()
    })
});