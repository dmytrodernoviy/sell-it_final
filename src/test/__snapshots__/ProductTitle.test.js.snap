// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle ProductTitle Link attribute 'to'  1`] = `ShallowWrapper {}`;
